src/WebJobs.Script/Description/Node/functions/functions.js

```diff
@@ -65,7 +69,12 @@ function createFunction(f) {
         inputs.unshift(context);
         delete context._inputs;
 
-        f.apply(null, inputs);
+        var userReturn = f.apply(null, inputs);
+        if (userReturn && typeof userReturn.then === 'function') {
```

Comment:
util.isFunction is deprecated, so I think the typeof check is a better choice.

As for promises and 'then': http://stackoverflow.com/a/27746324